src/components/molecules/Wallet/Details.tsx

```diff
@@ -1,16 +1,35 @@
-import React, { ReactElement } from 'react'
+import React, { ChangeEvent, ReactElement, useEffect, useState } from 'react'
 import Button from '../../atoms/Button'
 import styles from './Details.module.css'
 import { useOcean } from '@oceanprotocol/react'
 import Web3Feedback from './Feedback'
-import { getInjectedProviderName } from 'web3modal'
+import { getProviderInfo, IProviderInfo } from 'web3modal'
 import Conversion from '../../atoms/Price/Conversion'
 import { formatCurrency } from '@coingecko/cryptoformat'
 import { useUserPreferences } from '../../../providers/UserPreferences'
+import InputElement from '../../atoms/Input/InputElement'
 
 export default function Details(): ReactElement {
-  const { balance, connect, logout } = useOcean()
+  const { balance, connect, logout, web3Provider } = useOcean()
   const { locale } = useUserPreferences()
+  const [providerInfo, setProviderInfo] = useState<IProviderInfo>()
+  const [portisNetwork, setPortisNetwork] = useState<string>()
+
+  // Workaround cause getInjectedProviderName() always returns `MetaMask`
+  // https://github.com/oceanprotocol/market/issues/332
+  useEffect(() => {
+    if (!web3Provider) return
+
+    const providerInfo = getProviderInfo(web3Provider)
+    setProviderInfo(providerInfo)
+  }, [web3Provider])
+
+  // Handle network change for Portis
+  async function handlePortisNetworkChange(e: ChangeEvent<HTMLSelectElement>) {
+    setPortisNetwork(e.target.value)
+    web3Provider._portis.changeNetwork(e.target.value.toLowerCase())
+    await connect()
```

Comment:
@mihaisc something is happening in Portis with this implementation already but our `connect()` does not seem to work. Did you have some other idea here? Feel free to hack it in here

Reply:
ah, i think we need to refactor more